### ISSUE TITLE
master failing on Windows

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#4285]
Current master is failing because of 2 `cache` functionality tests.
I suspect the changes made in https://github.com/eslint/eslint/pull/4255 might have something to do with as it was working before.

Output:
```sh
  7047 passing (1m)
  2 failing

  1) CLIEngine executeOnFiles() cache should create the cache file inside the provided directory using the cacheLocation option:
     AssertionError: the cache for eslint was created: expected false to be true


  2) CLIEngine executeOnFiles() cache when the cacheFile is a directory or looks like a directory should create the cache file inside the provided dir
ectory:
     AssertionError: the cache for eslint was created: expected false to be true
```

**OS affected:** Windows 10 & 8

@royriojas anything you might know?

**Comment by royriojas** [comment]
@gyandeeps

Will try on a windows machine, not really sure what might be happening

Let me see.

**Comment by gyandeeps** [comment]
@royriojas Thanks for checking.

**Comment by royriojas** [comment]
I added the fix for windows. It should work now fine for windows paths.

**Comment by ilyavolodin** [comment]
Verified that the tests are no longer failing. Thanks @royriojas
